loopd/swapclient_server.go

```diff
@@ -992,9 +1002,50 @@ func validateLoopOutRequest(chainParams *chaincfg.Params, confTarget int32,
 	}
 
 	// Check that the label is valid.
-	if err := labels.Validate(label); err != nil {
+	if err := labels.Validate(req.Label); err != nil {
 		return 0, err
 	}
 
-	return validateConfTarget(confTarget, loop.DefaultSweepConfTarget)
+	channels, err := lnd.ListChannels(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	unlimitedChannels := len(req.OutgoingChanSet) == 0
+	outgoingChanSetMap := make(map[uint64]bool)
+	for _, c := range req.OutgoingChanSet {
+		outgoingChanSetMap[c] = true
+	}
+
+	var availabeBalance btcutil.Amount
+	for _, c := range channels {
+		// Don't bother adding the local balance of an inactive channel.
+		if !c.Active {
+			continue
+		}
+
+		// If no outgoing channel set was specified then all active
+		// channel local balances are considered. However, if a channel
+		// set was specified then only local balances of the specified
+		// channel are considered.
+		if unlimitedChannels || outgoingChanSetMap[c.ChannelID] {
+			availabeBalance += c.LocalBalance
+		}
+	}
+
+	// Determine if the total outgoing capacity of the channel set is
+	// enough to satisfy the loop amount and the maximum possible
+	// routing fees that could be charged.
+	if req.Amt+req.MaxSwapRoutingFee >
+		int64(availabeBalance.ToUnit(btcutil.AmountSatoshi)) {
```

Comment:
Could still be a bit simpler: 
```
requiredBalance := btcutil.Amount(req.Amt + req.MaxSwapRoutingFee)
if requiredBalance > availabeBalance{
  ...
}
```

Reply:
oops sorry. I did update that but must have undone it by accident. Will update asap